### PR TITLE
Remove WC v1

### PR DESCRIPTION
**Pull request by jinchung**
Fixes APP-1585

## What changed (plus any additional context for devs)
* Remove WalletConnect v1 (redux/walletconnect as well as entrypoints using WC v1)
* Remove requests redux (redux/requests) which was being used by both WC v1 and v2 and replace it with a new store (state/walletConnectRequests)
* Removed localstorage for WC requests
* We are choosing to ignore migration from redux/localstorage to the new store
* Follow up ticket created: alerting the user if they are using a WC v1 request

## Screen recordings / screenshots

https://github.com/user-attachments/assets/44d1e2ed-384a-448e-9aca-d48686505444

## What to test
* Test WC flows (message signing, transactions, connections) that they work as expected
* Test that requests still show up in the transaction list as expected and behave as expected when you interact with them

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -5,7 +5,6 @@ import { AppRegistry, Dimensions, LogBox, StyleSheet, View } from 'react-native'
 import { Toaster } from 'sonner-native';
 import { MobileWalletProtocolProvider } from '@coinbase/mobile-wallet-protocol-host';
 import { DeeplinkHandler } from '@/components/DeeplinkHandler';
-import { AppStateChangeHandler } from '@/components/AppStateChangeHandler';
```
Comment (jinchung) [inline src/App.tsx:8]: removing `AppStateChangeHandler` as it was being used only for WC v1 subscription when coming back from background

File: src/components/AppStateChangeHandler.tsx [hunk 2]
```diff
@@ -1,38 +0,0 @@
...
-  walletReady: boolean;
-};
-
-export function AppStateChangeHandler({ walletReady }: AppStateChangeHandlerProps) {
-  const [appState, setAppState] = useState(AppState.currentState);
-  const eventSubscription = useRef<ReturnType<typeof AppState.addEventListener> | null>(null);
-
-  const handleAppStateChange = useCallback(
-    (nextAppState: AppStateStatus) => {
-      if (appState === 'background' && nextAppState === 'active') {
```
Comment (jinchung) [inline src/components/AppStateChangeHandler.tsx:17]: no longer needed as this was for loading WC v1 when coming back from background app state

File: src/components/walletconnect-list/WalletConnectList.js [hunk 3]
```diff
@@ -1,44 +0,0 @@
...
-const scrollIndicatorInsets = {
-  bottom: scrollIndicatorInset,
-  top: scrollIndicatorInset,
-};
-
-const keyExtractor = item => item.dappUrl;
-
-const renderItem = ({ item }) => <WalletConnectListItem {...item} />;
-
-export default function WalletConnectList({ onLayout, ...props }) {
```
Comment (jinchung) [inline src/components/walletconnect-list/WalletConnectList.js:18]: unused component

File: src/hooks/useTransactionConfirmation.ts [hunk 4]
```diff
@@ -1,9 +0,0 @@
-import { removeRequest } from '../redux/requests';
-import { walletConnectSendStatus } from '../redux/walletconnect';
-
-export default function useTransactionConfirmation() {
```
Comment (jinchung) [inline src/hooks/useTransactionConfirmation.ts:4]: this was an unused hook

File: src/state/walletConnectRequests/index.ts [hunk 5]
```diff
@@ -0,0 +1,68 @@
+import { reverse, sortBy, values } from 'lodash';
+import { WalletconnectRequestData } from '@/walletConnect/types';
+import { createStore } from '../internal/createStore';
+import { omitFlatten } from '@/helpers/utilities';
+
+interface RequestsState {
+  [requestId: number]: WalletconnectRequestData;
+}
+
+export interface WalletConnectRequestsState {
```
Comment (jinchung) [inline src/state/walletConnectRequests/index.ts:10]: new store that replaces `redux/requests` and exposes the same methods for adding a request, removing a request, and getting a sorted list of the WC requests to display

File: src/walletConnect/index.tsx [hunk 6]
```diff
@@ -752,21 +752,9 @@ export async function onSessionRequest(event: SignClientTypes.EventArguments['se
       },
     };
 
-    const { requests: pendingRequests } = store.getState().requests;
```
Comment (jinchung) [inline src/walletConnect/index.tsx:755]: for the WC v2 flow, we used to directly update the `redux/requests` redux state as well as localstorage - now we use the new requests store

File: src/walletConnect/types.ts [hunk 7]
```diff
@@ -1,3 +1,115 @@
+import { ChainId } from '@/chains/types';
+import { Address } from 'viem';
+import { SignClientTypes, Verify } from '@walletconnect/types';
+import { RequestSource } from '@/utils/requestNavigationHandlers';
+
+/**
+ * Display details loaded for a request.
+ */
+interface RequestDisplayDetails {
```
Comment (jinchung) [inline src/walletConnect/types.ts:9]: moved some of the types from the `redux/requests` and `redux/walletconnect` over to this file instead

**Review by ibrahimtaveras00 — APPROVED** [review]
Looks good on both OS's and also did a sanity check on dapp browser to make sure we're good.  QA Passed 👍🏽

**Review by derHowie — APPROVED** [review]

File: src/components/asset-list/RecyclerAssetList2/index.tsx [hunk 8]
```diff
@@ -175,7 +173,7 @@ function NavbarOverlay({ accentColor, position }: { accentColor?: string; positi
       ].filter(Boolean),
       ...(ios ? { menuTitle: '' } : {}),
     }),
-    [activeWCV2Sessions.length, mostRecentWalletConnectors.length]
+    [activeWCV2Sessions.length]
```
Comment (greg-schrammel) [inline src/components/asset-list/RecyclerAssetList2/index.tsx:176]: why is this in the memo deps if not used in the menuConfig obj?

**Review by greg-schrammel — APPROVED** [review]
didn't test but the code is good 🫶

**Pull request merged**